service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -109,6 +121,55 @@ public void preparePipelineRun(
 
     // increment the prepare metric for this pipeline
     MetricsUtils.incrementPipelinePrepareRun(pipelineName);
+
+    return pipelineFileInputs;
+  }
+
+  /**
+   * Generate signed PUT urls and curl commands for each user-provided file input in the pipeline.
+   *
+   * <p>Each user-provided file input (assumed to be a path to a local file) is translated into a
+   * write-only (PUT) signed url in a location in the pipeline workspace storage container, in a
+   * directory defined by the jobId.
+   *
+   * <p>This signed url along with the source file path provided by the user are used to generate a
+   * curl command that the user can run to upload the file to the location in the pipeline workspace
+   * storage container.
+   */
+  private Map<String, Map<String, String>> prepareFileInputs(
+      Pipeline pipeline, UUID jobId, Map<String, Object> userProvidedInputs) {
+    // get the list of files that the user needs to upload
+    List<String> fileInputNames =
+        pipeline.getPipelineInputDefinitions().stream()
+            .filter(PipelineInputDefinition::getUserProvided)
+            .filter(p -> p.getType().equals(PipelineVariableTypesEnum.FILE))
+            .map(PipelineInputDefinition::getName)
+            .toList();
+
+    // TODO don't hardcode this
+    String projectId = "terra-dev-244bacba";
```

Comment:
store this value in db along with other workspace params? (this is specific to the workspace)

Reply:
ughhh the amountof things we're storing for a workspace is annoying.  Is this available through a terra api.  Maybe this is one of those we query for when we hit our Admin API

Reply:
yep, it is available via terra api (rawls: https://rawls.dsde-prod.broadinstitute.org/#/workspaces/listWorkspaceDetails you get workspace.googleProject and workspace.bucketName). i can make a ticket to retrieve both those things when you call admin endpoint

Reply:
https://broadworkbench.atlassian.net/browse/TSPS-307